arbnode/batch_poster.go

```diff
@@ -1322,11 +1322,30 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		return false, nil
 	}
 
-	lastPotentialMsg, err := b.streamer.GetMessage(msgCount - 1)
+
+	lastPotentialMsg, lastPotentialMsgInBatchPos, err := b.getLastPotentialMsgInBatch(msgCount)
+  atEspressoBatchLimit := b.building.msgCount >= *lastPotentialMsgInBatchPos
```

Comment:
Potential `nil` dereference?

Reply:
You're right I should have fixed it in [the commit](https://github.com/EspressoSystems/nitro-espresso-integration/pull/440/commits/4308c308422efff749652195dfd92ea6a53b71a4) to address your other comments.